### PR TITLE
print id detail when id is too long.

### PR DESCRIPTION
**Pull request by JaredTan95**
Print id detail when`id` is too long for easy debug.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/index/IndexRequest.java [hunk 1]
```diff
@@ -194,7 +194,7 @@ public ActionRequestValidationException validate() {
         validationException = DocWriteRequest.validateSeqNoBasedCASParams(this, validationException);
 
         if (id != null && id.getBytes(StandardCharsets.UTF_8).length > 512) {
-            validationException = addValidationError("id is too long, must be no longer than 512 bytes but was: " +
```
Comment (dnhatn) [inline server/src/main/java/org/elasticsearch/action/index/IndexRequest.java]: We prefer to include the id in square brackets. Can you change the message to "id [" + id + "] is too long ...".

**Review by dnhatn — COMMENTED** [review]
This change looks good. I left a minor suggestion. Can you also add a unit test for this change? Thank you.

File: server/src/main/java/org/elasticsearch/action/index/IndexRequest.java (hunk 1 above)
Comment (JaredTan95) [reply server/src/main/java/org/elasticsearch/action/index/IndexRequest.java]: Make sense to me. Glad to change it.

**Comment by dnhatn** [comment]
@elasticmachine ok to test

**Comment by dnhatn** [comment]
@elasticmachine update branch

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks @JaredTan95.

**Pull request merged**